core/utils/enums.py

```diff
@@ -0,0 +1,102 @@
+"""This file contains all the different enums used to implement callback
+pattern functionality."""
+
+from enum import Enum
+
+class JobTypes(Enum):
+    """
+    Enum to implement callback pattern for DMLRunner
+
+    JOB_TRAIN: Optimizer wants Runner to train the model
+
+    JOB_VAL: Optimizer wants Runner to validate the model
+
+    JOB_INIT: Optimizer wants Runner to initialize the model
+
+    JOB_AVG: Optimizer wants Runner to average weights via weighted avg.
+
+    JOB_COMM: Optimizer wants Runner to push the weights onto
+    IPFS and then communicate this to rest of nodes via blockchain
+    """
+    JOB_TRAIN = "TRAIN"
+    JOB_VAL = "VALIDATE"
+    JOB_INIT = "INITIALIZE"
+    JOB_AVG = "AVERAGE"
+    JOB_COMM = "COMMUNICATE"
+    NOTHING = "NOTHING"
+
+class RawEventTypes(Enum):
+    """
+    Enum to implement Communication Manager -> Optimizer callback pattern
+
+    The event indicates what has occurred previously in the Unix Service.
+
+    # NOTE: Following 3 will be replaced by MessageTypes in Gateway PR
+
+    NEW_SESSION: Received info from the blockchain to create a new 
+    decentralized learning session
+
+    NEW_WEIGHTS: Same as above, but for new weights
+
+    TERMINATE: Same as above, but to terminate
+
+    JOB_DONE: Sent to Communication Manager by Scheduler, means that
+    the payload will contain a DMLResult
+
+    NEW_INFO: Sent to Communication Manager by Gateway, means that
+    the payload contains info from the blockchain
+    """
+    NEW_SESSION = "NEW_SESSION"
+    NEW_WEIGHTS = "NEW_WEIGHTS"
+    TERMINATE = "TERMINATE"
+    JOB_DONE = "JOB_DONE"
+    NEW_INFO = "NEW_INFO"
+    NOTHING = "NOTHING"
```

Comment:
Is this realistic? If so, what's an example?

Reply:
idk removed

Reply:
Easy: 

1 - Create a mock Communication Manager and use that for the Scheduler's unit tests. Code:

```
class MockCommunicationManager:
    def inform(self, dummy1, dummy2):
        pass
```

2- (Bonus) Make the Scheduler have a `configure()` method where the Configuration Manager is passed. Let's actually make it a pattern that all connections between modules (dependency injections) should be done by a `configure()` method rather than the constructor. 

Now, in high level terms, what are realistic ways the Communication Manager will receive an event that says "NOTHING happened"?